src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -820,8 +820,7 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 						reflectionContext.AnalyzingPattern ();
 
 						var parameters = calledMethod.Parameters;
-						if ((parameters.Count == 3 && (methodParams[2].Kind == ValueNodeKind.MethodReturn || methodParams[2].Kind == ValueNodeKind.ConstInt || methodParams[2].Kind == ValueNodeKind.LoadField)
-							&& (methodParams[2].AsConstInt () == null || methodParams[2].AsConstInt () != 0)) ||
+						if ((parameters.Count == 3 && parameters[2].ParameterType.MetadataType == MetadataType.Boolean && (methodParams[2].AsConstInt () == null || methodParams[2].AsConstInt () != 0)) ||
```

Comment:
nit: instead of `methodParams[2].AsConstInt () == null || methodParams[2].AsConstInt () != 0` you could add something like `methodParams[2].IsDefault ()`

Reply:
While technically true I'm not sure it's worth it. The case here is specific - the parameter is a boolean and we're looking for `true` or `unknown`. Neither is "default" semantically.

Reply:
Actually it can be simplified as simple `!=0` - or the boolean equivalent `!=false`.